code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -142,6 +148,8 @@ var/global/list/animal_count = list() //Stores types, and amount of animals of t
 			stat = CONSCIOUS
 			density = 1
 			update_canmove()
+		if(canRegenerate == 1 && isRegenerating == 0)
```

Comment:
Since the values are just booleans doing `canRegenerate && !isRegenerating` is probably cheaper.